pandas/tests/indexes/test_base.py

```diff
@@ -1131,6 +1131,17 @@ def test_get_indexer_strings(self):
         with pytest.raises(TypeError):
             idx.get_indexer(['a', 'b', 'c', 'd'], method='pad', tolerance=2)
 
+    def test_get_indexer_consistency(self):
+        # See GH 16819
+        for name, index in self.indices.items():
+            indexer = index.get_indexer(index[0:2])
+            assert isinstance(indexer, np.ndarray)
```

Comment:
Issue with that is that some of the indexes are empty or categorical indexes so not unique positions so can't assume that [0, 1, 2] are returned